sympy/printing/latex.py

```diff
@@ -2506,31 +2506,52 @@ def _print_MatrixHomomorphism(self, h):
             self._print(h.domain), self._print(h.codomain))
 
     def _print_Manifold(self, manifold):
-        return r'\text{%s}' % manifold.name
+        string = manifold.name.name
+        if '{' in string:
+            name, supers, subs = string, [], []
+        else:
+            name, supers, subs = split_super_sub(string)
+
+            name = translate(name)
+            supers = [translate(sup) for sup in supers]
+            subs = [translate(sub) for sub in subs]
+
+        name = r'\text{%s}' % name
+        if supers:
+            name += "^{%s}" % " ".join(supers)
+        if subs:
+            name += "_{%s}" % " ".join(subs)
+
+        return name
 
     def _print_Patch(self, patch):
-        return r'\text{%s}_{\text{%s}}' % (patch.name, patch.manifold.name)
+        return r'\text{%s}_{%s}' % (self._print(patch.name), self._print(patch.manifold))
+
+    def _print_CoordSystem(self, coordsys):
+        return r'\text{%s}^{\text{%s}}_{%s}' % (
+            self._print(coordsys.name), self._print(coordsys.patch.name), self._print(coordsys.manifold)
+        )
 
-    def _print_CoordSystem(self, coords):
-        return r'\text{%s}^{\text{%s}}_{\text{%s}}' % (
-            coords.name, coords.patch.name, coords.patch.manifold.name
+    def _print_Point(self, point):
+        return '{%s}_{%s}' % (
+            self._print(point.coordinates()), self._print(point._coord_sys)
```

Comment:
I think that `_print_Point` should not be defined because it conflicts with printing of `Point` of `sympy.geometry` or `sympy.vector`. Maybe it was a bad idea to have duplicate names for classes at the first place, but it works better if you just leave it undefined and have some reasonable tuple notation of args.

Reply:
How about giving `diffgeom.Point` a new name, and make `'Point'` in diffgeom module as an aliasing?

Reply:
I have thought as a same idea to rename the Point in diffgeom. But it may be noted in the release notes.

Reply:
Deleted this. I think we can give different name to class and introduce aliasing, but that would better be handled in another PR.